### ISSUE TITLE
Assist: generate a `Default` impl from a `new` fn

### ISSUE THREAD

**Issue by yoshuawuyts** [rust-analyzer/rust-analyzer#7708]
_Sibling issue: https://github.com/rust-analyzer/rust-analyzer/issues/7709_

Clippy's ["new without default"](https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default) lint warns if you have a `new` method which doesn't take args, but don't have a `Default` impl. This is on by default. A nice assist would be if you hover over a `new` fn which takes no args that it suggests to generate a `Default` impl.

I'd be happy to implement this; feel free to assign me to it!

## Example
**before**
```rust
struct Example { _inner: () }

impl Example {
    pu|b fn new() -> Self {
        Self { _inner: () }
    }
}
```
**after**
```rust
struct Example { _inner: () }

impl Example {
    pub fn new() -> Self {
        Self { _inner: () }
    }
}

impl Default for Example {
    fn default() -> Self {
        Self::new()
    }
}
```

**Comment by chetankhilosiya** [comment]
FYI I am working on this.

**Comment by lnicola** [comment]
Implemented in #7800
